app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java

```diff
@@ -0,0 +1,440 @@
+                    mViewDirection.setRotation((float) Math.toDegrees(mViewPLayer.getPanoramaPositionAngle()));
+                }
+            };
+
+            @Override
+            public void run() {
+                while (mIsThreadPositionActive) {
+                    try {
+                        Thread.sleep(50);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                    mViewDirection.post(mRunnableUpdateRotation);
+                }
+            }
+        };
+        mThreadUpdatePosition.start();
+
+    }
+
+    private void stopThreadUpdateRotation() {
+        mIsThreadPositionActive = false;
+        if (mThreadUpdatePosition != null)
+            mThreadUpdatePosition.interrupt();
+        mThreadUpdatePosition = null;
+    }
+
+    private void updateViewPlayPause() {
+        if (mPlayer.getPlayWhenReady())
+            mBtnPlayPause.setImageResource(R.drawable.ic_player_pause);
+        else
+            mBtnPlayPause.setImageResource(R.drawable.ic_player_play);
+    }
+
+    private void toast(String message) {
+        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
+
+
+    }
+
+    private void onVariantSelected(int position) {
+        if (mVariants.get(position).isAuto()) {
+            mVariantAdapter.setSelected(position);
+            mPlayer.setAutoVariant();
+        } else {
+            mVariantAdapter.setSelected(position);
+            mPlayer.setCurrentVariant(mVariants.get(position).getIndex());
+        }
+        mVariantAdapter.notifyDataSetChanged();
+    }
+
+    private void setDisplayMode() {
+        if (mViewPLayer.getState() == IBandPlayerView.DisplayState.EQUIRECTANGULAR && mStream.getState() == IBandStream.State.OPEN) {
+            if (mViewPLayer.getVRMode()) {
+                mBottomActionBar.setVisibility(View.GONE);
+                mBtnPlayPause.setVisibility(View.GONE);
+                mLvSettings.setVisibility(View.GONE);
+                mBtnVrMode.setVisibility(View.GONE);
+                mBtnCenter.setVisibility(View.GONE);
+            } else {
+                mBottomActionBar.setVisibility(View.VISIBLE);
+                mBtnPlayPause.setVisibility(View.VISIBLE);
+                mBtnVrMode.setVisibility(View.VISIBLE);
+                mBtnCenter.setVisibility(View.VISIBLE);
+            }
+        }
+    }
+
+    private void initPlayer() {
+        mIBandSDK = new IBandSDK(this);
+        mStream = mIBandSDK.createStream(STREAM_ID);
+        mStream.addListener(this);
+        mPlayer = mIBandSDK.createPlayer();
+        mPlayer.addListener(this);
+        mPlayer.setStream(mStream);
+        if (mStream.getType() == IBandStream.Type.LIVE)
+            mViewLive.setVisibility(View.VISIBLE);
+        else
+            mViewLive.setVisibility(View.GONE);
+    }
+
+    private void initViews() {
+        mViewPLayer = (IBandPlayerView) findViewById(R.id.view_player);
```

Comment:
This function is too long.
Split it to partial short methods.

Reply:
done